sensor/common/registry/registry_store.go

```diff
@@ -334,3 +350,76 @@ func (rs *Store) GetMatchingCentralRegistryIntegrations(imgName *storage.ImageNa
 
 	return regs
 }
+
+// UpsertImageContentSourcePolicy will store a new/updated ImageContentSourcePolicy.
+func (rs *Store) UpsertImageContentSourcePolicy(icsp *operatorV1Alpha1.ImageContentSourcePolicy) {
+	rs.mirrorSetMutex.Lock()
+	defer rs.mirrorSetMutex.Unlock()
+
+	rs.icspRules[string(icsp.GetUID())] = icsp
+}
+
+// DeleteImageContentSourcePolicy will delete an ImageContentSourcePolicy from the store if it exists.
+func (rs *Store) DeleteImageContentSourcePolicy(uid string) {
+	rs.mirrorSetMutex.Lock()
+	defer rs.mirrorSetMutex.Unlock()
+
+	delete(rs.icspRules, uid)
+}
+
+// UpsertImageDigestMirrorSet will store a new/updated ImageDigestMirrorSet.
+func (rs *Store) UpsertImageDigestMirrorSet(idms *configV1.ImageDigestMirrorSet) {
+	rs.mirrorSetMutex.Lock()
+	defer rs.mirrorSetMutex.Unlock()
+
+	rs.idmsRules[string(idms.GetUID())] = idms
+}
+
+// DeleteImageDigestMirrorSet will delete an ImageDigestMirrorSet from the store if it exists.
+func (rs *Store) DeleteImageDigestMirrorSet(uid string) {
+	rs.mirrorSetMutex.Lock()
+	defer rs.mirrorSetMutex.Unlock()
+
+	delete(rs.idmsRules, uid)
+}
+
+// UpsertImageTagMirrorSet will store a new/updated ImageTagMirrorSet.
+func (rs *Store) UpsertImageTagMirrorSet(itms *configV1.ImageTagMirrorSet) {
+	rs.mirrorSetMutex.Lock()
+	defer rs.mirrorSetMutex.Unlock()
+
+	rs.itmsRules[string(itms.GetUID())] = itms
+}
+
+// DeleteImageTagMirrorSet will delete an ImageTagMirrorSet from the store if it exists.
+func (rs *Store) DeleteImageTagMirrorSet(uid string) {
+	rs.mirrorSetMutex.Lock()
+	defer rs.mirrorSetMutex.Unlock()
+
+	delete(rs.itmsRules, uid)
+}
+
+// GetAllMirrorSets returns slices of all the stored mirror sets as expected by openshift/runtime-utils.
+//
+// ref: https://github.com/openshift/runtime-utils/blob/5c488b20a19fc8c1fee9011c41ce70379bc8ca4d/pkg/registries/registries.go#L240
+func (rs *Store) GetAllMirrorSets() ([]*operatorV1Alpha1.ImageContentSourcePolicy, []*configV1.ImageDigestMirrorSet, []*configV1.ImageTagMirrorSet) {
+	rs.mirrorSetMutex.RLock()
+	defer rs.mirrorSetMutex.RUnlock()
+
+	icspRules := make([]*operatorV1Alpha1.ImageContentSourcePolicy, 0, len(rs.icspRules))
+	for _, rule := range rs.icspRules {
+		icspRules = append(icspRules, rule)
+	}
+
+	idmsRules := make([]*configV1.ImageDigestMirrorSet, 0, len(rs.idmsRules))
+	for _, rule := range rs.idmsRules {
+		idmsRules = append(idmsRules, rule)
+	}
+
+	itmsRules := make([]*configV1.ImageTagMirrorSet, 0, len(rs.itmsRules))
+	for _, rule := range rs.itmsRules {
+		itmsRules = append(itmsRules, rule)
+	}
+
+	return icspRules, idmsRules, itmsRules
```

Comment:
There's `maputil.Values()` which can be used to simplify things here:
```suggestion
	return maputil.Values(rs.icspRules), maputil.Values(rs.idmsRules), maputil.Values(rs.itmsRules)
```